### PR TITLE
RabbitMQ Weak Credentials Plugin

### PR DESCRIPTION
**Pull request by leonardo-doyensec**
The following PR adds the weak credential tester for RabbitMQ Management Portal.

Some notes on the current version of the rabbitmq management portal credential tester.

- does a custom fingerprint in the `canAccept()` method since rabbitmq is not correctly recognized from nmap
- tests the credentials using Basic Authentication
- the management portal needs to be exposed on a different port, because the default one, which is 15672, is not scanned by the default configuration of nmap

### CODE REVIEW

**Review by tooryx — COMMENTED** [review]
Thank you @leonardo-doyensec! This looks good. Added a minor change request.

_~tooryx_

**Comment by leonardo-doyensec** [comment]
Hi @tooryx,
i've applied the fix that you have requested.

Thank you,
Leonardo

**Comment by tooryx** [comment]
Thank you! I will try to get this merged as soon as possible.

_~tooryx_

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import javax.inject.Inject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+/** Credential tester for RabbitMQ Management Portal. */
+public final class RabbitMQCredentialTester extends CredentialTester {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final HttpClient httpClient;
+
+  private static final String RABBITMQ_SERVICE = "rabbitmq";
```
Comment (tooryx) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java:47]: I missed it before but this variable is not used anymore given that we are not relying on the nmap service.

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    try {
+      JsonObject response = JsonParser.parseString(responseBody).getAsJsonObject();
+
+      if (response.has("name") && response.has("tags")) {
+        logger.atInfo().log("Successfully logged in to RabbitMQ Management Portal");
+        return true;
+      } else {
+        return false;
+      }
+    } catch (Exception e) {
```
Comment (tooryx) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java:194]: This catch block catches `Exception`, but can only catch unchecked exceptions. Consider catching RuntimeException (or something more specific) instead so it is more apparent that no checked exceptions are being handled.

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  private static String buildTargetUrl(NetworkService networkService, String path) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    targetUrlBuilder.append(path);
+    return targetUrlBuilder.toString();
+  }
+
+  // Checks if the response body contains the title element of rabbitmq management page.
+  // Custom fingerprint phase.
+  private static boolean bodyContainsRabbitMQElements(String responseBody) {
```
Comment (tooryx) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java:81]: Minor style issue (Unfortunately I cannot fix it myself): Methods and non-static variables should be named in lowerCamelCase, with acronyms treated as words 

```diff
-  private static boolean bodyContainsRabbitMQElements(String responseBody) {
+ private static boolean bodyContainsRabbitMqElements(String responseBody) {

{...}

- .map(RabbitMQCredentialTester::bodyContainsRabbitMQElements)
+ .map(RabbitMQCredentialTester::bodyContainsRabbitMqElements)
```

File: google/detectors/credentials/generic_weak_credential_detector/src/test/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTesterTest.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  private static final TestCredential WEAK_CRED_1 =
+      TestCredential.create("user", Optional.of("1234"));
+  private static final TestCredential WEAK_CRED_2 =
+      TestCredential.create("root", Optional.of("pass"));
+  private static final TestCredential WRONG_CRED_1 =
+      TestCredential.create("wrong", Optional.of("pass"));
+
+  private static final String WEAK_CRED_AUTH_1 = "Basic dXNlcjoxMjM0";
+  private static final String WEAK_CRED_AUTH_2 = "Basic cm9vdDpwYXNz";
+  private static final ServiceContext.Builder RABBITMQ_SERVICE_CONTEXT =
```
Comment (tooryx) [inline google/detectors/credentials/generic_weak_credential_detector/src/test/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTesterTest.java:63]: The builder is mutable, so please consider renaming it `rabbitmqServiceContext`

**Review by tooryx — COMMENTED** [review]
Sorry I just came across a few style issues. I would normally fix it myself, but I do not have access to the doyensec repo. Could you please take a look?

_~tooryx_

**Comment by leonardo-doyensec** [comment]
No problem, i've fixed them.

Leonardo

**Pull request merged**